scripts/functions-deploy-tests/cli.ts

```diff
@@ -0,0 +1,56 @@
+import * as spawn from "cross-spawn";
+import { ChildProcess } from "child_process";
+
+interface Result {
+  proc: ChildProcess;
+  stdout: string;
+  stderr: string;
+}
+
+/**
+ * Execute a Firebase CLI command.
+ */
+export function exec(
+  cmd: string,
+  project: string,
+  additionalArgs: string[],
+  cwd: string,
+  quiet = true
+): Promise<Result> {
```

Comment:
I'm still in the side of "this is duplicate code" and should be consolidated. If combining them is a larger task than would be reasonable to do in this PR, let's put a TODO in place and file a bug for ourselves to come back to this.